### ISSUE TITLE
dialog.showMessageBox with Callback blocking the process

### ISSUE THREAD

**Issue by hacdias** [electron/electron#17801]
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
  * v4.1.4
* **Operating System:**
  * macOS 10.14.4
* **Last Known Working Electron version:**:
  * N/A

### Expected Behavior

`dialog.showMessageBox` would not block the process since it has got a `callback`. It's stated in the [docs](https://electronjs.org/docs/api/dialog#dialogshowmessageboxbrowserwindow-options-callback) the following:

> If a callback is passed, the dialog will not block the process. The API call will be asynchronous and the result will be passed via callback(response).

### Actual Behavior

The process gets blocked.

### To Reproduce

```
import { app, dialog } from 'electron'

async function run () {
    await app.whenReady()

    dialog.showMessageBox({
      message: 'Title',
      detail: 'Detail',
      buttons: ['Yes', 'No']
    }, option => {
      console.log(option)
    })
    
    console.log('Got it.')
}

run()
```

### Screenshots

N/A

### Additional Information

N/A

**Comment by MarshallOfSound** [comment]
This is, unfortunately, a documentation issue that I fixed in

https://github.com/electron/electron/commit/e01c3615c49bd1b02f550f71332313a667b4b352#diff-fb9981169251617b6c87eb2a78549c97

But it appears that that information was lost after the promisification.

TLDR: Is that you need to provide the `browserWindow` parameter for it to be truly async.

**Comment by hacdias** [comment]
Thanks for the quick response @MarshallOfSound 😄 I'll see what I can do then.

**Comment by hacdias** [comment]
@MarshallOfSound I was looking at master branch's docs and saw that there is/will be a `showMessageBox` and a `showMessageBoxAsync`. Do you have any idea of when it will be available?

**Comment by monokh** [comment]
@MarshallOfSound I find that the statement in your commit is not accurate. It's only on Mac that a browserWindow is required. On Linux, it will be asynchronous with only the callback. I find it odd that this is inconsistent so it's a bug in my view.
